parachain/integration-test/src/lib.rs

```diff
@@ -0,0 +1,379 @@
+		Arc::new(chain_a_sub_client.clone()) as Arc<dyn IsmpProvider>,
+		Arc::new(chain_b_sub_client.clone()) as Arc<dyn IsmpProvider>,
+	);
+
+	let mut client_map = HashMap::new();
+	client_map
+		.insert(chain_a_sub_client.clone().state_machine_id().state_id, chain_a_client.clone());
+	client_map
+		.insert(chain_b_sub_client.clone().state_machine_id().state_id, chain_b_client.clone());
+
+	let (client_a, client_b) = (chain_a_sub_client.clone().client, chain_b_sub_client.client);
+
+	relay(
+		chain_a_sub_client,
+		chain_b_client.clone(),
+		relayer_config.clone(),
+		Kusama(3000), // random coprocessor id
+		tx_payment,
+		client_map.clone(),
+		&task_manager,
+	)
+	.await?;
+
+	// Accounts & keys
+	let alice_signer = PairSigner::<Hyperbridge, _>::new(
+		Pair::from_string("//Alice", None).expect("Unable to create ALice account"),
+	);
+	let alice_key = api::storage().system().account(AccountId32(dev::alice().public_key().0));
+
+	let amount = 100_000_000000000000;
+	let transfer_call = api::tx().ismp_demo().transfer(TransferParams {
+		to: AccountId32(dev::alice().public_key().0),
+		amount,
+		para_id: 2001,
+		timeout: 70,
+	});
+
+	let alice_chain_a_initial_balance = client_a
+		.storage()
+		.at_latest()
+		.await?
+		.fetch(&alice_key)
+		.await?
+		.ok_or("Failed to fetch")
+		.unwrap()
+		.data
+		.free;
+
+	let alice_chain_b_initial_balance = client_b
+		.storage()
+		.at_latest()
+		.await?
+		.fetch(&alice_key)
+		.await?
+		.ok_or("Failed to fetch")
+		.unwrap()
+		.data
+		.free;
+
+	let result = client_a
+		.tx()
+		.sign_and_submit_then_watch_default(&transfer_call, &alice_signer)
+		.await?
+		.wait_for_finalized_success()
+		.await?
+		.all_events_in_block()
+		.clone();
+
+	let tx_block_hash = result.block_hash();
+
+	let events = client_a.events().at(tx_block_hash).await?;
+	log::info!("Ismp Events: {:?} \n", events.find_last::<Request>()?);
+
+	// Assert burnt & transferred tokens in chain A
+	let alice_chain_a_new_balance = client_a
+		.storage()
+		.at_latest()
+		.await?
+		.fetch(&alice_key)
+		.await?
+		.ok_or("Failed to fetch")
+		.unwrap()
+		.data
+		.free;
+
+	// Time delay for the relayer to submit the request to the dest chain
+	tokio::time::sleep(Duration::from_secs(40)).await;
+
+	// The relayer should finish sending the request message to chain B
+
+	let alice_chain_b_new_balance = client_b
+		.storage()
+		.at_latest()
+		.await?
+		.fetch(&alice_key)
+		.await?
+		.ok_or("Failed to fetch")
+		.unwrap()
+		.data
+		.free;
+
+	// diving by 10000000000 for better assertion as the rem balance = initial - amount - fees
+	// in chain A
+	assert_eq!(
+		(alice_chain_a_initial_balance - amount) / 100000000000,
+		alice_chain_a_new_balance / 100000000000
+	);
+	// in chain B
+	assert_eq!(
+		(alice_chain_b_initial_balance + amount) / 100000000000,
+		alice_chain_b_new_balance / 100000000000
+	);
+
+	Ok(())
+}
+
+/// fetch a foreign storage item from a given key
+#[tokio::test(flavor = "multi_thread")]
```

Comment:
Same as above

```suggestion
#[tokio::test(flavor = "multi_thread")]
#[ignore]
```

also add these tests to the github workflow